### PR TITLE
Create Frame in FrameConnection

### PR DESCRIPTION
**Pull request by cesarbs**

### CODE REVIEW

File: src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/FrameConnection.cs [hunk 1]
```diff
@@ -90,11 +90,17 @@ public void StartRequestProcessing()
...
+                    ConnectionId = _context.ConnectionId,
+                    ConnectionInformation = _context.ConnectionInformation,
+                    ServiceContext = _context.ServiceContext,
+                    TimeoutControl = this,
+                    Input = input,
+                    Output = new OutputProducer(output, ConnectionId, Log)
+                });
+
+                // Don't yield control to the transport until we've added the connection to the connection manager
                 _context.ServiceContext.ConnectionManager.AddConnection(_context.FrameConnectionId, this);
```
Comment (halter73) [inline src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/FrameConnection.cs]: Maybe add a comment that `_frame` must be initialized prior to calling this.

File: test/Microsoft.AspNetCore.Server.Kestrel.Core.Tests/FrameResponseHeadersTests.cs [hunk 2]
```diff
@@ -61,14 +63,14 @@ public void InitialDictionaryIsEmpty()
...
-        [InlineData("Unknown�-Header", "Data")]
-        [InlineData("Ser�ver", "Data")]
+        [InlineData("Ser™ver", "Data")]
+        [InlineData("Server", "Da™ta")]
+        [InlineData("Unknown™-Header", "Data")]
+        [InlineData("Ser™ver", "Data")]
+        [InlineData("šerver", "Data")]
+        [InlineData("Server", "Dašta")]
+        [InlineData("Unknownš-Header", "Data")]
+        [InlineData("Seršver", "Data")]
```
Comment (halter73) [inline test/Microsoft.AspNetCore.Server.Kestrel.Core.Tests/FrameResponseHeadersTests.cs]: Wasn't this file already converted to UTF-8?

Comment (cesarbs) [inline test/Microsoft.AspNetCore.Server.Kestrel.Core.Tests/FrameResponseHeadersTests.cs]: In my other PR 😄

File: src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/FrameContext.cs [hunk 3]
```diff
@@ -10,5 +12,8 @@ public class FrameContext
         public string ConnectionId { get; set; }
         public ServiceContext ServiceContext { get; set; }
         public IConnectionInformation ConnectionInformation { get; set; }
+        public ITimeoutControl TimeoutControl { get; set; }
+        public IPipeReader Input { get; set; }
+        public OutputProducer Output { get; set; }
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/FrameContext.cs]: Make this `IPipe`

File: src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/FrameConnection.cs [hunk 4]
```diff
@@ -90,11 +90,19 @@ public void StartRequestProcessing()
...
-                _frame.Output = new OutputProducer(output, ConnectionId, Log);
+                // _frame must be initialized before adding the connection to the connection manager
+                _frame = new Frame<TContext>(application, new FrameContext
+                {
+                    ConnectionId = _context.ConnectionId,
+                    ConnectionInformation = _context.ConnectionInformation,
+                    ServiceContext = _context.ServiceContext,
+                    TimeoutControl = this,
+                    Input = input,
+                    Output = new OutputProducer(output, ConnectionId, Log)
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/FrameConnection.cs]: Set this in the Frame ctor

**Pull request merged**